gems/aws-crt/spec/spec_helper.rb

```diff
@@ -22,5 +22,6 @@ def garbage_collect_is_immediate?
 def check_for_clean_shutdown
   ObjectSpace.garbage_collect
 
-  Aws::Crt::Native.thread_join_all_managed
+  # TODO: call new FFI version of below
+  # Aws::Crt::Native.thread_join_all_managed
```

Comment:
Is this still necessary? This should be handled by shutting down all of the CRT libraries

Reply:
I'm not sure - The original intent of this was for use in tests to make sure that Ruby code was not holding onto native resources somewhere: "If this fails, there is a reference somewhere keeping these resources alive."  I'm not sure whether that is still useful or not.  @graebm may know.

Reply:
Yup, it's still a good idea to call this.
This function *ought to* raise an exception if it couldn't join all the threads (because someone somewhere was keeping something alive).
If we call it at the end of each test, we'll see which test is leaking resources

When this is called during library shutdown by C code, it doesn't check the error code. So there's no alert if something is still alive. Also, if we only checked at library shutdown we wouldn't know exactly which test was responsible for the leak